Runtime/MeshOperations/CombineMeshes.cs

```diff
@@ -114,6 +114,196 @@ public static List<ProBuilderMesh> Combine(IEnumerable<ProBuilderMesh> meshes)
             return res;
         }
 
+        /// <summary>
+        /// Merge a collection of <see cref="ProBuilderMesh"/> objects to as few meshes as possible. It will re-use the meshTarget object as the first
+        /// destination for the first 65535 vertices. If the sum of vertices is above 65535 it will generate new meshes unless there is a single mesh left in which
+        /// case it will append it to the return list.
+        /// </summary>
+        /// <param name="meshTarget">A polybrush mesh for which we will be re-using the target mesh of the previous.</param>
+        /// <param name="meshes">A collection of meshes to be merged.</param>
+        /// <returns>
+        /// A list of merged meshes. In most cases this will be a single mesh corresponding to meshTarget. However it can be multiple in cases
+        /// where the resulting vertex count exceeds the maximum allowable value.
+        /// </returns>
+        public static List<ProBuilderMesh> Combine(ProBuilderMesh meshTarget, IEnumerable<ProBuilderMesh> meshes)
```

Comment:
I would suggest to re-use `List<ProBuilderMesh> Combine(IEnumerable<ProBuilderMesh> meshes)` logic then inject the result into a given target. To me, the logic stays the same, only where we store the first mesh data changes (and we still generate new ProBuilderMeshes if we exceed the  vertex count limit).

Reply:
It is not totally the same as we need to convert the other mesh instances into the first target space and we do not want to alter the target internal data as we are processing those, which minimizes the manipulation. Furthermore, if we exceed the vertex count we are using the old method, except in the case if there is a single mesh remaining to merge which would fail the call(I am unsure in this case what would be the benefit of reprocessing it to generate a new mesh?)

Am I missing your point?
Thank you!

Reply:
I see what you mean but I feel the code has some redundant parts and could probably be divided better or refactored. For example, I would have maybe taken the opportunity to offer a new function to Combine a mesh as much as it can with a list of meshes within the limit of vertex count like if you were creating a texture atlas. Then use `List<ProBuilderMesh> Combine (IEnumerable<ProBuilderMesh> meshes)` to combine the remaining meshes like you did at the end.

On a side note, I also don't see the benefit with having the first loop if the sole idea is to inject the meshTarget data  into `vertices`, `sharedVertices`, `sharedTextures` and `materialMap`. Maybe I'm missing something. Otherwise, I would go with `FirstOrDefault()` to check if the object exists in the passed collection.

Maybe what I'm proposing is too overkill. @karl- , thoughts?

Reply:
@thomas-tu  I definitely agree with your comment on the first loop. Will unloop.

Reply:
I agree that we should not have two separate functions for combining meshes.

My thoughts on this are we should make the behavior of keeping the first source mesh gameobject and all it's components the default, removing the current path.